### ISSUE TITLE
Radio Button high contrast hover state not visible when contained in a selected item.

### ISSUE THREAD

**Issue by StephenLPeters** [microsoft/microsoft-ui-xaml#2647]
Copied from internal [issue](https://microsoft.visualstudio.com/OS/_workitems/edit/23065239?src=WorkItemMention&src-action=artifact_link).

When a radio button exists in an item which is selected, like a list view item, the hovered state is not visible in high constrast mode.

![Screenshot (114)](https://user-images.githubusercontent.com/7758786/84437697-4b2a9400-abea-11ea-81e8-c7bdb7354227.png)

Some design work is needed to determine the proper high contrast hover state style.

Similar to #2561

**Comment by StephenLPeters** [comment]
@chigy fyi

**Comment by chigy** [comment]
I believe the same design model as what we are doing to fix the checkbox intermediate checkbox design would work here as well. (border and middle circle white, pressed becomes all white)

![image](https://user-images.githubusercontent.com/22958680/84960863-af9c9600-b0b7-11ea-9704-7fabf1925d36.png)

**Comment by Felix-Dev** [comment]
I would like to look into this.

@chigy To clarify, would the look of the RadioButton change as well the mouse hovers above it? If I'm reading your comment correctly, you've outlined the diresired visual look of the RadioButton when in rest state and pressed state. That leaves the hover state.

**Comment by chigy** [comment]
@Felix-Dev , hover state is there in the above table. I've specified Rest/Hover/Pressed. I did not specify disabled because I was not sure if disabled/selected is a thing...

**Comment by Felix-Dev** [comment]
@chigy I don't think we will be able to use the TreeViewItem checkbox HC solution here for generically used RadioButtons. To recap, here's the issue we are attempting fix:

![radio-button-hc-issue](https://user-images.githubusercontent.com/1398851/85219112-3647c080-b3a1-11ea-8e7d-71f7c96e4565.gif)

![radio-button-hc-issue-black](https://user-images.githubusercontent.com/1398851/85219139-7c048900-b3a1-11ea-999a-57aff829ba73.gif)

If I understood you correctly with your diagram above RadioButton [mouse hover] should change the border to white in HC White. But you can already see the problem with that as now the RadioButton won't be visible any longer when in rest state with mouse hover:

![radio-button-hc-issue-white-with-whitehoverborder](https://user-images.githubusercontent.com/1398851/85219327-0b5e6c00-b3a3-11ea-8662-a6934bdd30ab.gif)

In the [TreeViewItem checkbox case](https://github.com/microsoft/microsoft-ui-xaml/issues/2561), we can set the border to white on mouse hover because we can *control* the appaearance of the checkbox in *relation* to the background of the TreeViewItem (by using the TreeViewItem's visual states). However, in our case here we don't know what color the background of the panel is where the RadioButton has been placed in. For example, in the image shared by @StephenLPeters it seems likely the RadioButton is inside a DataTemplate used to define the look of a ListViewItem. We as WinUI have no control over that. In case of the TreeViewItem's checkbox we do though because that's a WinUI control.

Let me know if I misinterpreted your diagram for the RadioButton hover state but from my current understanding we still need to find woking hover state style for the RadioButton to work when used both on a black/white background and an accent colored background as shown in the images above.

PS. Another issue with the current HC color theming of the RadioButton is that [mouse hover] and [click] (when RadioButton is unchecked) are indistinguishable. Here's a GIF showing the problem:

![radio-button-hc-issue-black-click](https://user-images.githubusercontent.com/1398851/85220723-56ca4780-b3ae-11ea-993c-5c248c162d5a.gif)

Your diagram above makes sure these two states can be distinguished from each other. Now we only need to find a a color style which works for the [mouse hover] state as outlined above.

**Comment by chigy** [comment]
@Felix-Dev , I see what you mean. This has been a known issue on HC because we do not have control over the background color developer decides to use. In this case, it feels almost we need two styles that can be switched by the developers who knows if the background works with white version or colored versions. Does this sound correct?

**Comment by JuanMejiaVelez** [comment]
One thought I had with this issue is to get the Boxing that happens around the Content of the RadioButton be extended to the ellipse so that it is guarded from the background color change. Is that possible?

**Comment by YuliKl** [comment]
@chigy @marksfoster this is likely to be resolved in the near future with updated HC colors. Would be good to keep an eye on this problem and close the issue when appropriate.
